### PR TITLE
Feature/performance enhancement

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
## PR Overview
**This PR will address the following Issue/Feature:**
- #117 
- #125 
- Possibly others

**This PR will result in the following new package version:**

- v0.17.0: materialization changes

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

> ## 🚨 Breaking Changes 🚨
> > ⚠️ Since the following changes are breaking, a `--full-refresh` after upgrading will be required.
> - To reduce storage, updated default materialization of the staging models to views. 
> 
> ## Performance improvements
>   - Updated the incremental strategy of the following models to `insert_overwrite` for BigQuery and Databricks All Purpose Cluster destinations and `delete+insert` for all other supported destinations. 
>     - `int_jira__issue_calendar_spine`
>     - `int_jira__pivot_daily_field_history`
>     - `jira__daily_issue_field_history`
>     > At this time, models for Databricks SQL Warehouse destinations are materialized as tables without support for incremental runs.
>   - Removed intermediate models `int_jira__agg_multiselect_history`, `int_jira__combine_field_histories`, and `int_jira__daily_field_history` by combining them with `int_jira__pivot_daily_field_history`. This is to reduce the redundancy of the data stored in tables, the number of full scans, and the volume of write operations.
>   - Updated the default materialization of `int_jira__issue_type_parents` from a table to a view. This model is used in one downstream model, so a view will reduce storage requirements while not significantly hindering performance.
>   - For Snowflake and BigQuery destinations, added `cluster_by` columns to the configs for incremental models.
>   - For Databricks All Purpose Cluster destinations, updated incremental model file formats to `parquet` for compatibility with the `insert_overwrite` strategy.
> 
> ## Features
> - Added a default 3-day look-back to incremental models to accommodate late arriving records. The number of days can be changed by setting the var `lookback_window` in your dbt_project.yml. See the [Lookback Window section of the README](https://github.com/fivetran/dbt_jira/blob/main/README.md#lookback-window) for more details.
> 
> ## Under the Hood:
> - Added integration testing pipeline for Databricks SQL Warehouse.
> - Updated the maintainer pull request template.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [x] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

- See internal ticket

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 packages:
-- package: fivetran/jira_source
-  version: [">=0.6.0", "<0.7.0"]
\ No newline at end of file
+# - package: fivetran/jira_source
+#   version: [">=0.6.0", "<0.7.0"]
+  - git: https://github.com/fivetran/dbt_jira_source.git
+    revision: feature/performance-enhancement
+    warn-unpinned: false
```
Comment (fivetran-catfritz) [inline packages.yml:6]: Update before merge
```suggestion
 - package: fivetran/jira_source
   version: [">=0.7.0", "<0.8.0"]
```

**Comment by fivetran-catfritz** [comment]
Will regen docs after final approval

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz Great work on this PR!! I just have a few comments to address. Overall though this PR is great and passed all my stress tests!! Once you have made the final updates and regen the docs then this should be good to be approved!

File: models/intermediate/field_history/int_jira__issue_calendar_spine.sql [hunk 2]
```diff
@@ -73,6 +73,12 @@ issue_spine as (
         and {{ dbt.dateadd('month', var('jira_issue_history_buffer', 1), 'issue_dates.open_until') }} >= spine.date_day
         -- if we cut off issues, we're going to have to do a full refresh to catch issues that have been un-resolved
 
+    {% if is_incremental() %}
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/intermediate/field_history/int_jira__issue_calendar_spine.sql:76]: Yes this is intentional. The first part is not used to find new records, but is used to limit the initial size of the spine using `min(earliest_open_until_date)` but with an adequate range for backfilling before the final `max(date_day)` filter, so no lookback is necessary.

File: models/intermediate/int_jira__issue_type_parents.sql [hunk 3]
```diff
@@ -1,8 +1,4 @@
-{{
-    config(
-        materialized='table'
-    )
-}}
+{{ config(materialized='view') }}
```
[the remark this answers is not in the document]
Comment (fivetran-catfritz) [reply models/intermediate/int_jira__issue_type_parents.sql:1]: It used to be a table, but needs to be at least a view to use the dbt_utils.star macro in int_jira__issue_users.

**Review by fivetran-catfritz — COMMENTED** [review]
Thanks @fivetran-joemarkiewicz. Updated and ready for re-review!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM! Great work on this PR @fivetran-catfritz

**Pull request merged**